modules/components/tag-input/tag-input.ts

```diff
@@ -509,11 +509,8 @@ export class TagInputComponent extends TagInputAccessor implements OnInit, After
      * @name appendTag
      * @param tag {TagModel}
      */
-    public appendTag = (tag: TagModel): void => {
-        this.items = [
-            ...this.items,
-            this.modelAsStrings ? tag[this.identifyBy] : tag
-        ];
+    public appendTag = (tag: TagModel, index = this.items.length): void => {
+        this.items.splice(index, 0, tag);
```

Comment:
sorry, but doing so this.items gets mutated, I'd like something immutable instead